src/screens/BookByFilterScreen.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {useEffect, useState} from 'react';
+import {
+  StyleSheet,
+  SafeAreaView,
+  FlatList,
+  Text,
+  TouchableOpacity,
+} from 'react-native';
+import {bookType} from './HomeScreen';
+import {getBook} from '../api/callApi';
+import {CardBooks} from '../components/CardBook';
+import {HeaderDefault} from '../components/HeaderDefault';
+
+const BookByFilterScreen = (props: {route: {params: {filterName: string}}}) => {
+  const [books, setBooks] = useState<bookType[]>([]);
+  const [examMode, setExamMode] = useState<boolean>(false);
+
+  useEffect(() => {
+    getBook().then(res => {
+      setBooks(
+        res.data.viewer.books.hits
+          .filter((book: bookType) => {
+            const {levels, subjects} = book;
+            const filterName = props.route.params.filterName;
+            return (
+              (levels &&
+                levels.some((level: any) => level.name === filterName)) ||
+              (subjects && subjects.some((sub: any) => sub.name === filterName))
+            );
+          })
+          .sort((a: bookType, b: bookType) => {
+            const aLevels = a.levels || [];
+            const bLevels = b.levels || [];
+            const aLevelName = aLevels.length > 0 ? aLevels[0].name : '';
+            const bLevelName = bLevels.length > 0 ? bLevels[0].name : '';
+            return aLevelName.localeCompare(bLevelName);
+          }),
+      );
+    });
+  }, [props.route.params.filterName]);
+
+  return (
+    <SafeAreaView style={{flex: 1}}>
+      <HeaderDefault title={props.route.params.filterName} />
+      <TouchableOpacity
+        style={{alignSelf: 'center', margin: 15}}
+        onPress={() => {
+          setExamMode(!examMode);
```

Comment:
Using the callback version of the state setter is more robust.

```suggestion
          setExamMode(prevExamMode => !prevExamMode);
```